test/repos/tasks/test_sync_zoo_file.py

```diff
@@ -0,0 +1,160 @@
+from typing import Dict, List, Union
+
+import pytest
+
+from zoo.repos import tasks as uut
+from zoo.repos.models import Repository
+from zoo.repos.zoo_yml import parse
+from zoo.services.models import Environment, Service
+
+pytestmark = pytest.mark.django_db
+
+
+@pytest.fixture()
+def generate_repositories(repository_factory):
+    repository_factory(
+        id=1,
+        remote_id=11,
+        name="test_proj1",
+        owner="john_doe1",
+        url="https://github.com/john_doe1/test_proj1",
+        provider="github",
+    )
+
+
+def test_update_or_create_service(generate_repositories):
+    data = {
+        "type": "service",
+        "name": "test_proj1",
+        "owner": "john_doe1",
+        "impact": "profit",
+        "status": "beta",
+        "docs_url": "http://test_proj1/docs",
+        "slack_channel": "#test_proj1",
+        "sentry_project": "http://test_proj1/sentry",
+        "sonarqube_project": "http://test_proj1/sonarqube",
+        "pagerduty_url": "http://test_proj1/pager_duty",
+        "tags": ["tag1", "tag2", "tag3"],
+        "environments": [
+            {
+                "name": "staging",
+                "dashboard_url": "http://staging.test_proj1/dashboard",
+                "service_urls": [
+                    "http://staging.test_proj1/service1",
+                    "http://staging.test_proj1/service2",
+                ],
+                "health_check_url": "http://staging.test_proj1/health_check",
+            },
+            {
+                "name": "production",
+                "dashboard_url": "http://production.test_proj1/dashboard",
+                "service_urls": [
+                    "http://production.test_proj1/service1",
+                    "http://production.test_proj1/service2",
+                ],
+                "health_check_url": "http://production.test_proj1/health_check",
+            },
+        ],
+    }
+
+    proj = {"id": 11, "provider": "github"}
+
+    uut.update_or_create_service(data, proj)
+    try:
+        service = Service.objects.filter(owner=data["owner"], name=data["name"]).first()
```

Comment:
`first()` will never trigger an error, you'd have to use `get(owner=..., name=...)` I think.